p2p/conn/secret_connection.go

```diff
@@ -249,17 +255,22 @@ func (sc *SecretConnection) Read(data []byte) (n int, err error) {
 	defer pool.Put(frame)
 	_, err = sc.recvAead.Open(frame[:0], sc.recvNonce[:], sealedFrame, nil)
 	if err != nil {
-		return n, fmt.Errorf("failed to decrypt SecretConnection: %w", err)
+		return n, ErrDecryptConnection{source: err}
```

Comment:
The error here is because we failed to decrypt a "frame", not the connection.

Reply:
Done.

Reply:
resolved